### PR TITLE
feat: replace lib imports by importlib

### LINKED ISSUE #191: Deprecation warning: the imp module is deprecated in favour of importlib
Hello.
There is a deprecation warning coming from config.py at this line:

https://github.com/mtth/hdfs/blob/8984b0a4f6510b3a4c8880c3da0553d6b2d36b32/hdfs/config.py#L15

```
../../../../usr/local/lib/python3.9/site-packages/hdfs/config.py:15
/usr/local/lib/python3.9/site-packages/hdfs/config.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
from imp import load_source
```

According to the [official documentation](https://docs.python.org/3/library/imp.html): 

_Deprecated since version 3.4, will be removed in version 3.12: The [imp](https://docs.python.org/3/library/imp.html#module-imp) module is deprecated in favor of [importlib](https://docs.python.org/3/library/importlib.html#module-importlib)._

-----

`Python 3.9`
`HdfsCLI 2.6.0` and `2.7.0`

### PR DESCRIPTION
**Pull request by Andromelus**
Goal: fix #191 
I am using an alternate laptop for this and cannot run the tests. 😞

### CODE REVIEW

**Comment by tjni** [comment]
I was looking at this too and noticed that `importlib` is not available in Python 2.7 so the import needs to be conditional if that version still needs to be supported.

**Disclaimer:** I'm not affiliated with this project so can't facilitate this getting in.

**Comment by Andromelus** [comment]
@tjni Indeed, that may be usefull. I'll have a look into it this week.

Concerning the build fail, checking on Internet, seems like Python 3.6 is not available for the latest version of Ubuntu.

**Comment by Andromelus** [comment]
@tjni The test framework only supports Python 3, not bellow. Therefore I will keep the tests on python 3.6 and 3.12.

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+Changes
+
+* Remove references to Python 2 from documentation
+  * The test library only support Python 3
+* Bump CI github actions Ubuntu runner version from `latest` to `20.04`
+* Prepare Ci github actions to launch the tests against Python 3.12
+  * `fastavro` does not support Python 3.12 yet
+* Use conditionnal imports for ``lib``/``importlib`` to prepare for library migration in Python 3.12
+  * More informations [here](https://docs.python.org/3/library/imp.html)
+
```
Comment (mtth) [inline CHANGES:16]: These are useful to maintainers but not end users, let's omit for now. We can advertise 3.12 support once it's tested in CI.

File: hdfs/config.py [hunk 2]
```diff
@@ -21,6 +20,12 @@
 import os.path as osp
 import sys
 
+try:
+  # python 3.12 and above
+  from importlib import load_source
+except ImportError:
+  # Bellow python 3.12
```
Comment (mtth) [inline hdfs/config.py:27]: Typo in "below" and "Python" (should be capitalized). Same in `test_examples.py`.

**Review by mtth — CHANGES_REQUESTED** [review]
Thank you for the PR @Andromelus and apologies for the very belated review. This `importlib` change sounds good. A few comments inline.

**Review by mtth — APPROVED** [review]
Thanks for the update @Andromelus, looks good.

**Comment by musicinmybrain** [comment]
This is still broken; there is no `importlib.load_source` in Python 3.12 either. One other project handled it like this: https://github.com/return42/fspath/commit/f09288e1cb0744b760d2221efa670af2eb17e927

**Pull request merged**